src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -153,6 +155,12 @@ private async Task<FormulaValue> TryHandleSet(CallNode node, SymbolContext conte
         public override async ValueTask<FormulaValue> Visit(CallNode node, SymbolContext context)
         {
             CheckCancel();
+
+            //Don't know if this is the best place to put this increment call.
+            if (!context.IncrementCallDepth(_maxRecursionDepth))
```

Comment:
So, this depth check is specific to UDFs with recursion (not impl yet). You'd want to increment it specifically when entering evaluation of a UDF.